examples/systemcalls/systemcalls.c

```diff
@@ -92,6 +135,56 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+	pid_t pid = fork();
```

Comment:
@akka2103 think about how you could avoid duplicating logic in these two functions.